### PR TITLE
Add support in Joiner for input stages with a combination of macro an…

### PR DESCRIPTION
**Pull request by rmstar**
…d non-macro output schemas.

### CODE REVIEW

File: core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java [hunk 1]
```diff
@@ -383,6 +378,12 @@ Schema getOutputSchema(Map<String, Schema> inputSchemas, FailureCollector collec
     return outputFields;
   }
 
+  private boolean inputSchemasAvailable(Map<String, Schema> inputSchemas, List<String> inputStages) {
+    return inputSchemas.keySet().containsAll(inputStages) &&
+      inputSchemas.values().stream().noneMatch(Objects::isNull);
+
```
Comment (albertshau) [inline core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java]: nit: remove empty line

File: pom.xml [hunk 2]
```diff
@@ -678,7 +678,7 @@
           <version>1.1.0</version>
           <configuration>
             <cdapArtifacts>
-              <parent>system:cdap-data-pipeline[6.1.0-SNAPSHOT,7.0.0-SNAPSHOT)</parent>
+              <parent>system:cdap-data-pipeline[6.2.0-SNAPSHOT,7.0.0-SNAPSHOT)</parent>
               <parent>system:cdap-data-streams[6.1.0-SNAPSHOT,7.0.0-SNAPSHOT)</parent>
```
Comment (albertshau) [inline pom.xml]: should also update the lower bound for this

**Review by albertshau — APPROVED** [review]
minor comments, otherwise lgtm. 

Please be sure the build (https://builds.cask.co/browse/HYP-BAD674) is green before merging. It looks like it's currently red due to some unrelated failure, so I kicked off another run

File: core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java (hunk 1 above)
Comment (rmstar) [reply core-plugins/src/main/java/io/cdap/plugin/batch/joiner/Joiner.java]: fixed.

File: pom.xml (hunk 2 above)
Comment (rmstar) [reply pom.xml]: fixed.

**Pull request merged**